osu.Game/Overlays/NowPlayingOverlay.cs

```diff
@@ -469,5 +471,98 @@ protected override void OnHoverLost(HoverLostEvent e)
                 base.OnHoverLost(e);
             }
         }
+
+        private partial class ScrollingTextContainer : CompositeDrawable
+        {
+            private const float initial_move_delay = 1000;
+            private const float pixels_per_second = 50;
+
+            private LocalisableString text;
+            private OsuSpriteText mainSpriteText = null!;
+            private OsuSpriteText fillerSpriteText = null!;
+
+            public LocalisableString Text
+            {
+                get => text;
+                set
+                {
+                    text = value;
+                    Schedule(updateText);
+                }
+            }
+
+            public FontUsage Font
+            {
+                set =>
+                    Schedule(() =>
+                    {
+                        mainSpriteText.Font = value;
+                        fillerSpriteText.Font = value;
+
+                        updateText();
+                    });
+            }
+
+            public ScrollingTextContainer()
+            {
+                AutoSizeAxes = Axes.Both;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load()
+            {
+                InternalChild = new FillFlowContainer<OsuSpriteText>
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Direction = FillDirection.Horizontal,
+                    Children = new[]
+                    {
+                        mainSpriteText = new OsuSpriteText { Padding = new MarginPadding { Horizontal = margin } },
+                        fillerSpriteText = new OsuSpriteText { Padding = new MarginPadding { Horizontal = margin }, Alpha = 0 },
+                    }
+                };
+            }
+
+            private void updateText()
+            {
+                mainSpriteText.Text = text;
+                fillerSpriteText.Alpha = 0;
+
+                ClearTransforms();
+                X = 0;
+
+                float textOverflowWidth = mainSpriteText.Width - player_width;
+
+                if (textOverflowWidth > 0)
+                {
+                    fillerSpriteText.Alpha = 1;
+                    fillerSpriteText.Text = text;
+
+                    float initialX;
+                    float targetX;
+
+                    if (Anchor.HasFlagFast(Anchor.x0))
+                    {
+                        initialX = 0;
+                        targetX = -mainSpriteText.Width;
+                    }
+                    else if (Anchor.HasFlagFast(Anchor.x1))
+                    {
+                        initialX = (textOverflowWidth + mainSpriteText.Width) / 2;
+                        targetX = (textOverflowWidth - mainSpriteText.Width) / 2;
+                    }
+                    else // Anchor.x2
+                    {
+                        initialX = textOverflowWidth + mainSpriteText.Width;
+                        targetX = textOverflowWidth;
+                    }
```

Comment:
no visual tests for this? how do I check how different anchors work if the only usage is centred?

also I'm not even entirely sure how right-anchored text scrolling should work, as in should it still really be scrolling to the left?

Reply:
If it's always centered do we even need support for other anchors? This is a very localised usage so that feels counter-intuitive.

Reply:
Have removed for now.